admin/app/components/solidus_admin/roles/edit/component.html.erb

```diff
@@ -5,6 +5,36 @@
         <%= render component("ui/forms/field").text_field(f, :name, class: "required") %>
         <%= render component("ui/forms/field").text_field(f, :description) %>
       </div>
+
+      <% if permission_set_options.present? %>
+        <h3 class="text-l py-4 font-semibold text-gray-900">
+          <%= t(".choose_permissions") %>
+        </h3>
+      <% end %>
+
+      <%= permission_set_component(form: f, title: t('.orders'), permissions: permission_set_options[:order]) %>
+      <%= permission_set_component(form: f, title: t('.products'), permissions: permission_set_options[:product]) %>
+      <%= permission_set_component(form: f, title: t('.stock'), permissions: permission_set_options[:stock]) %>
+      <%= permission_set_component(form: f, title: t('.customers'), permissions: permission_set_options[:user]) %>
+      <%= permission_set_component(form: f, title: t('.restricted_stock'), permissions: permission_set_options[:restricted_stock]) %>
+      <%= permission_set_component(form: f, title: t('.settings'), permissions: permission_set_options[:configuration]) %>
+
+      <% if permission_set_options[:other].present? %>
+        <div class="hover:bg-gray-50 rounded-md">
```

Comment:
I think a shared component would make sense to reduce the amount of HTML (and nasty Tailwind classes) we need to maintain in the future?

Reply:
Shared component `CheckboxRow` has been added as it encapsulates the entirety of the repeated HTML, and not just the checkbox itself, good suggestion thanks!